benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/scheduler/ExperimentTaskScheduler.java

```diff
@@ -15,43 +15,50 @@
 import cloud.benchflow.experimentmanager.tasks.running.ReExecuteTrialTask;
 import cloud.benchflow.experimentmanager.tasks.running.execute.ExecuteTrial.TrialStatus;
 import cloud.benchflow.experimentmanager.tasks.start.StartTask;
-
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Future;
-
+import java.util.concurrent.LinkedBlockingDeque;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * @author Jesper Findahl (jesper.findahl@usi.ch) created on 2017-04-19
  */
-public class ExperimentTaskController {
+public class ExperimentTaskScheduler {
 
   private static Logger logger =
-      LoggerFactory.getLogger(ExperimentTaskController.class.getSimpleName());
+      LoggerFactory.getLogger(ExperimentTaskScheduler.class.getSimpleName());
 
   private ConcurrentMap<String, Future> experimentTasks = new ConcurrentHashMap<>();
 
-  // TODO - running queue (1 element)
+  // ready queue
+  private BlockingQueue<String> readyQueue = new LinkedBlockingDeque<>();
 
-  // TODO - ready queue
+  // running queue (1 element)
+  private BlockingQueue<String> runningQueue = new ArrayBlockingQueue<>(1, true);
 
   private BenchFlowExperimentModelDAO experimentModelDAO;
   private TrialModelDAO trialModelDAO;
   // TODO - should go into a stateless queue (so that we can recover)
   private ExecutorService experimentTaskExecutorService;
   private BenchFlowTestManagerService testManagerService;
 
-  public ExperimentTaskController(ExecutorService experimentTaskExecutorService) {
+  public ExperimentTaskScheduler(ExecutorService experimentTaskExecutorService) {
 
     this.experimentTaskExecutorService = experimentTaskExecutorService;
 
     this.experimentModelDAO = BenchFlowExperimentManagerApplication.getExperimentModelDAO();
     this.trialModelDAO = BenchFlowExperimentManagerApplication.getTrialModelDAO();
     this.testManagerService = BenchFlowExperimentManagerApplication.getTestManagerService();
+
+    // start dispatcher in separate thread
+    new Thread(new ExperimentDispatcher(readyQueue, runningQueue)).start();
```

Comment:
Or share the queues with the package modifier, so that all the classes in the `scheduler` package can access the queues (that I think it should be because we should only have classes needing the queues in the `scheduler` package). Then it it fine that they are initialised by the main actor in the package, that is the `ExperimentTaskScheduler`, that is always the first class that get instantiated in the main application. This way you can initialise also the dispatcher in the main application class.